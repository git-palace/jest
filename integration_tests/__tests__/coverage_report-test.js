/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @emails oncall+jsinfra
 */
'use strict';

const {linkJestPackage} = require('../utils');
const runJest = require('../runJest');
const fs = require('fs');
const path = require('path');

const DIR = path.resolve(__dirname, '../coverage_report');

beforeEach(() => linkJestPackage('babel-jest', DIR));

it('outputs coverage report', () => {
  const {stdout, status} = runJest(DIR, ['--no-cache', '--coverage']);
  const coverageDir = path.resolve(__dirname, '../coverage_report/coverage');

  // - the `setup.js` file is ignored and should not be in the coverage report.
  // - `sum_dependency.js` is mocked and the real module is never required but
  //  is listed with 0 % coverage.
  // - `not-required-in-test-suite.js` is not required but it is listed
  //  with 0 % coverage.
  expect(stdout).toMatchSnapshot();

  expect(() => fs.accessSync(coverageDir, fs.F_OK)).not.toThrow();
  expect(status).toBe(0);
});

it('collects coverage only from specified files', () => {
  const {stdout} = runJest(DIR, [
    '--no-cache',
    '--coverage',
    '--collectCoverageFrom', // overwrites the one in package.json
    'setup.js',
  ]);

  // Coverage report should only have `setup.js` coverage info
  expect(stdout).toMatchSnapshot();
});
