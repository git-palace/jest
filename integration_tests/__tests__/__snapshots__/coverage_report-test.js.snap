exports[`test collects coverage only from specified files 1`] = `
"----------|----------|----------|----------|----------|----------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
----------|----------|----------|----------|----------|----------------|
All files |      100 |      100 |      100 |      100 |                |
 setup.js |      100 |      100 |      100 |      100 |                |
----------|----------|----------|----------|----------|----------------|
"
`;

exports[`test json reporter printing with --coverage 1`] = `
"Coverage Summary (add \"text\" to the \"coverageReporters\" setting to receive a full report)
 › Statements: 100%
 › Branches: 100%
 › Lines: 100%
 › Functions: 100%
"
`;

exports[`test outputs coverage report 1`] = `
"-------------------------------|----------|----------|----------|----------|----------------|
File                           |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-------------------------------|----------|----------|----------|----------|----------------|
All files                      |    45.45 |        0 |       50 |    45.45 |                |
 not-required-in-test-suite.js |        0 |        0 |        0 |        0 | 10,17,21,22,24 |
 other-file.js                 |      100 |      100 |      100 |      100 |                |
 sum.js                        |      100 |      100 |      100 |      100 |                |
 sum_dependency.js             |        0 |      100 |      100 |        0 |             10 |
-------------------------------|----------|----------|----------|----------|----------------|
"
`;
